src/main/java/in/projecteka/consentmanager/common/Role.java

```diff
@@ -0,0 +1,9 @@
+package in.projecteka.consentmanager.common;
+
+public enum Role {
+    GATEWAY;
+
+    public static Role valueOfIgnoreCase(String mayBeRole) {
+         return  mayBeRole.equalsIgnoreCase(GATEWAY.name()) ? GATEWAY : null;
+    }
```

Comment:
Isn't it supposed to return Optional\<Role>?

Reply:
Why can't it return just Role?